seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/utils/JdbcCatalogUtils.java

```diff
@@ -409,4 +428,102 @@ private static ReadonlyConfig extractCatalogConfig(JdbcConnectionConfig config)
         catalogConfig.put(JdbcOptions.HANDLE_BLOB_AS_STRING.key(), config.isHandleBlobAsString());
         return ReadonlyConfig.fromMap(catalogConfig);
     }
+
+    /** Process table path with regex pattern and add matched tables to the result. */
+    private static void processRegexTablePath(
+            AbstractJdbcCatalog jdbcCatalog,
+            JdbcDialect jdbcDialect,
+            JdbcSourceTableConfig tableConfig,
+            Map<TablePath, JdbcSourceTable> result)
+            throws SQLException {
+
+        String tablePath = tableConfig.getTablePath();
+        log.info("Processing table path with regex: {}", tablePath);
+
+        // Parse table path to extract database, schema and table patterns
+        String databasePattern = ".*";
+        String tableNamePattern = ".*"; // This is just the table name part
+
+        // Convert the table path to a format suitable for regex matching
+
+        // Step 1: Replace escaped dots (\.) with a placeholder
+        String processedTablePath = tablePath.replace("\\.", DOT_PLACEHOLDER);
+        log.info("After replacing escaped dots with placeholder: {}", processedTablePath);
+
+        // Step 2: Split by unescaped dots
+        String[] parts = processedTablePath.split("\\.");
+
+        String fullTablePattern;
+
+        if (parts.length == 1) {
+            // Only table pattern
+            tableNamePattern = parts[0];
+            fullTablePattern = String.format("%s", tableNamePattern);
+        } else if (parts.length == 2) {
+            // database.table or schema.table format
+            databasePattern = parts[0];
+            tableNamePattern = parts[1];
+            fullTablePattern = String.format("%s.%s", databasePattern, tableNamePattern);
+        } else if (parts.length >= 3) {
+            // database.schema.table format
+            databasePattern = parts[0];
+            String schemaPattern = parts[1];
+            tableNamePattern = parts[2];
+            fullTablePattern =
+                    String.format(
+                            "%s.%s.%s",
+                            databasePattern.replace(DOT_PLACEHOLDER, "."),
+                            schemaPattern.replace(DOT_PLACEHOLDER, "."),
+                            tableNamePattern.replace(DOT_PLACEHOLDER, "."));
+        } else {
+            fullTablePattern = ".*..*..*";
```

Comment:
```suggestion
            fullTablePattern = ".*";
```
Can this work properly?